src/cascadia/TerminalApp/CascadiaSettings.cpp

```diff
@@ -569,58 +555,66 @@ TerminalSettings CascadiaSettings::BuildSettings(GUID profileGuid) const
 // - the GUID of the profile corresponding to this combination of index and NewTerminalArgs
 GUID CascadiaSettings::_GetProfileForArgs(const NewTerminalArgs& newTerminalArgs) const
 {
-    std::optional<int> profileIndex{ std::nullopt };
-    if (newTerminalArgs &&
-        newTerminalArgs.ProfileIndex() != nullptr)
+    std::optional<GUID> profileByIndex, profileByName;
+    if (newTerminalArgs)
     {
-        profileIndex = newTerminalArgs.ProfileIndex().Value();
+        if (newTerminalArgs.ProfileIndex() != nullptr)
+        {
+            profileByIndex = _GetProfileGuidByIndex(newTerminalArgs.ProfileIndex().Value());
+        }
+
+        profileByName = _GetProfileGuidByName(newTerminalArgs.Profile());
     }
-    GUID profileGuid = _GetProfileForIndex(profileIndex);
 
-    if (newTerminalArgs)
-    {
-        const auto profileString = newTerminalArgs.Profile();
+    return Utils::CoalesceOptionals(profileByName, profileByIndex, _globals.GetDefaultProfile());
+}
 
-        // First, try and parse the "profile" argument as a GUID. If it's a
-        // GUID, and the GUID of one of our profiles, then use that as the
-        // profile GUID instead. If it's not, then try looking it up as a
-        // name of a profile. If it's still not that, then just ignore it.
-        if (!profileString.empty())
+// Method Description:
+// - Helper to get the GUID of a profile given a name that could be a guid or an actual name.
+// Arguments:
+// - name: a guid string _or_ the name of a profile
+// Return Value:
+// - the GUID of the profile corresponding to this name
+std::optional<GUID> CascadiaSettings::_GetProfileGuidByName(const std::wstring_view name) const

```

Comment:
nit: The name of this function hides the fact that a GUID is acceptable. Maybe rename it to just `_GetProfileGuid()`?

I was going to say `GetProfileGuidByString` but the fact that it takes in a string makes the name kinda redundant.

Reply:
Side note: maybe you'd want to rename `GetProfileGuidByIndex` too to make them look nicer together but eh, I don't really care about that one.